src/sonic-config-engine/portconfig.py

```diff
@@ -72,12 +73,12 @@ def db_connect_configdb(namespace=None):
     """
     Connect to configdb
     """
-    try:
-        if namespace is not None:
+    if not swsscommon.SonicDBConfig.isInit():
+        if multi_asic.is_multi_asic():
             swsscommon.SonicDBConfig.load_sonic_global_db_config(namespace=namespace)
```

Comment:
>load_sonic_global_db_config [](http://example.com/codeflow?start=37&length=27)

portconfig.py is just a library. It's the application's responsibility to load db config. Applications include:
1. sonic-cfggen
2. config
3. etc